src/util/codec/mysql/json/functions.rs

```diff
@@ -0,0 +1,163 @@
+    let mut ret = String::with_capacity(s.len());
+    let mut chars = s.chars();
+    while let Some(ch) = chars.next() {
+        if ch == '\\' {
+            let c = match chars.next() {
+                Some(c) => c,
+                None => return Err(box_err!("Incomplete escaped sequence")),
+            };
+            match c {
+                '"' => ret.push('"'),
+                'b' => ret.push(CHAR_BACKSPACE),
+                'f' => ret.push(CHAR_FORMFEED),
+                'n' => ret.push(CHAR_LINEFEED),
+                'r' => ret.push(CHAR_CARRIAGE_RETURN),
+                't' => ret.push(CHAR_HORIZONTAL_TAB),
+                '\\' => ret.push('\\'),
+                'u' => {
+                    let mut unicode = String::with_capacity(ESCAPED_UNICODE_BYTES_SIZE);
+                    for _ in 0..ESCAPED_UNICODE_BYTES_SIZE {
+                        match chars.next() {
+                            Some(c) => unicode.push(c),
+                            None => return Err(box_err!("Invalid unicode: {}", unicode)),
+                        }
+                    }
+                    let utf8 = try!(decode_escaped_unicode(&unicode));
+                    ret.push(utf8);
+                }
+                _ => ret.push(c),
+            }
+        } else {
+            ret.push(ch);
+        }
+    }
+    Ok(ret)
+}
+
+fn decode_escaped_unicode(s: &str) -> Result<char> {
+    let u = box_try!(u32::from_str_radix(s, 16));
+    char::from_u32(u).ok_or(box_err!("invalid char from: {}", s))
+}
+
+#[cfg(test)]
+mod test {
+    use std::collections::BTreeMap;
+    use super::*;
+
+    #[test]
+    fn test_decode_escaped_unicode() {
+        let mut test_cases = vec![
+            ("5e8a", '床'),
+            ("524d", '前'),
+            ("660e", '明'),
+            ("6708", '月'),
+            ("5149", '光'),
+        ];
+        for (i, (escaped, expected)) in test_cases.drain(..).enumerate() {
+            let d = decode_escaped_unicode(escaped);
+            assert!(d.is_ok(), "#{} expect ok but got err {:?}", i, d);
+            let got = d.unwrap();
+            assert_eq!(got,
+                       expected,
+                       "#{} expect {:?} but got {:?}",
+                       i,
+                       expected,
+                       got);
+        }
+    }
+
+    #[test]
+    fn test_json_unquote() {
+        // test unquote json string
+        let mut test_cases = vec![("\\b", true, Some("\x08")),
+                                  ("\\f", true, Some("\x0C")),
+                                  ("\\n", true, Some("\x0A")),
+                                  ("\\r", true, Some("\x0D")),
+                                  ("\\t", true, Some("\x09")),
+                                  ("\\\\", true, Some("\x5c")),
+                                  ("\\u597d", true, Some("好")),
+                                  ("0\\u597d0", true, Some("0好0")),
+                                  ("[", true, Some("[")),
```

Comment:
add a `\\a` here, and it should be unquoted as `a`, which is compatible with MySQL document, but different from its implement.